### ISSUE TITLE
Quantulum incorrect for prefixes that are not capitilized if the unit is incorrectly capitilized

### ISSUE THREAD

**Issue by hwalinga** [nielstron/quantulum3#161]
**Describe the bug**
When a unit is incorrectly capitilized, the SI-prefix will be read wrong. 

**To Reproduce**

```py
from quantulum3 import parser

parser.parse('1mw')  # returns: Megawatt
parser.parse('1pw')  # returns: Petawatt
```

**Expected behavior**

```py
from quantulum3 import parser

parser.parse('1mw')  # should be: Milliwatt
parser.parse('1pw')  # should be: Picowatt
```

**Note**
`pW` and `PW`, `mW` and `MW` are correctly read. 

**Additional information:**
 - Python Version: [3.7, 3.8]
 - Classifier activated/ sklearn installed: yes (`pw` will not be read if not installed)
 - OS: MX Linux 19 (Debian 10 derivative)
 - Version: 0.7.5

**Comment by nielstron** [comment]
I see the issue, the main problem being that for no match given the actual capitalization of the input, the closest match for all-lower input is given (completely ignoring capitalization).

The most consistent way around this would be to return the first match obtained with as little capitalization changes as necessary (potentially preferring changing from lower to upper over upper to lower).
However this would mean that 2^(len(units)) changes are evaluated in the worst case. However this may not be a problem as unit strings do not contain more than 4 seperate units

**Comment by hwalinga** [comment]
Why not leave capitalization intact for matching a no-match object? Especially for prefixes this is a huge difference.

**Comment by nielstron** [comment]
What do you mean by "leave capitalization intact"? If units were compared without changing capitalization, "mw" would neither match "MW" nor "mW" which are the only valid official versions of writing Mega- or milliwatt respectively.

Switching to all-lower, all of "mw" is matched to "MW" as well as "mW" which both are valid units. I see that another option would be to give preference to the match that is closest to the original string (which would return mW here, as expected)

**Comment by hwalinga** [comment]
Then I don't know exactly how quantulum works. I thought there was machine learning doing the matches, not exact matching? 

Also, if you for example change from scikit-learn 0.23.2 to the older 0.23.1 the results change. 

This is 0.23.1: 

```py
>>> from quantulum3 import parser
>>>
>>> parser.parse('1mw')
[Quantity(1, "Unit(name="milliwatt", entity=Entity("power"), uri=Milliwatt)")]
>>> parser.parse('1pw')
[Quantity(1, "Unit(name="picowatt", entity=Entity("power"), uri=Picowatt)")]
>>>
>>> parser.parse('1Pw')
[Quantity(1, "Unit(name="picowatt", entity=Entity("power"), uri=Picowatt)")]
```

Now, only the last one is not correct.

**Comment by nielstron** [comment]
It depends. Most of the tool is a very complex regex-based mechanism, it will turn up with "1pw" or "1mw". The unit in the back is then compared to all known units. If multiple units match, either a simple heuristic or the machine learning part is used to decide which unit has the highest likelihood of being correct.

Hmm random output sounds bad. On the exact same input? Once trained, the output of the trained network should be deterministic.

**Comment by hwalinga** [comment]
Yes, random output is bad. At least a warning should be thrown if picking a random option. Anyway, I made a PR.
